### PR TITLE
Add direct file upload

### PR DESCRIPTION
**Pull request by mariobodemann**
Implement Contentful API for [directly uploading files](https://www.contentful.com/developers/docs/references/content-management-api/#/reference/uploads).

### CODE REVIEW

File: src/main/java/com/contentful/java/cma/ModuleUploads.java [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+import static java.util.Arrays.copyOf;
+
+/**
+ * Upload Module.
+ * <p>
+ * This module will take care of all `upload.contentful.com` related issues, as in directly
+ * uploading a file to contentful, receiving it's id.
+ */
+public final class ModuleUploads extends AbsModule<ServiceUploads> {
+  private static final String CONTENT_TYPE_STREAM = "application/octet-stream";
```
Comment (Roblinde) [inline src/main/java/com/contentful/java/cma/ModuleUploads.java]: Here it's called CONTENT_TYPE_STREAM, what's the difference between this and OCTET_STREAM_CONTENT_TYPE?

File: src/main/java/com/contentful/java/cma/ModuleUploads.java [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+   * @throws java.io.IOException      if the stream could not be read.
+   */
+  public CMAUpload create(String spaceId, InputStream stream) throws IOException {
+    assertNotNull(spaceId, "spaceid");
+    assertNotNull(stream, "stream");
+
+    final byte[] content = readAllBytes(stream);
+
+    final RequestBody payload = RequestBody.create(MediaType.parse(CONTENT_TYPE_STREAM), content);
+    return service.create(spaceId, payload).toBlocking().first();
```
Comment (Roblinde) [inline src/main/java/com/contentful/java/cma/ModuleUploads.java]: excuse my Java, but why .toBlocking()? Do we want to block the thread here?

File: src/main/java/com/contentful/java/cma/ModuleUploads.java (hunk 1 above)
Comment (mariobodemann) [inline src/main/java/com/contentful/java/cma/ModuleUploads.java]: That is a very good question. I'll align it to take the same name.

File: src/main/java/com/contentful/java/cma/ModuleUploads.java (hunk 2 above)
Comment (mariobodemann) [inline src/main/java/com/contentful/java/cma/ModuleUploads.java]: No worries: In this case: Yes we want the thread to block. If someone would like to use it async. then they'll want to use `client.`*`async().`*`create(...)`

**Pull request merged**